### PR TITLE
refactor: network status for stardust

### PR DESCRIPTION
**Pull request by nicole-obrien**
## Summary
Please summarize your changes, describing __what__ they are and __why__ they were made.

### Changelog
```
Please write an accurate and concise changelog for your changes.
```

## Relevant Issues
Please list any related issues (e.g. bug, task).

## Type of Change
Please select any type below that is applicable to your changes, and delete those that are not.
- [ ] __Breaking__ - any change that would cause existing functionality to not work as expected
- [ ] __Chore__ - refactoring, build scripts or anything else that isn't user-facing
- [ ] __Docs__ - changes to the documentation
- [ ] __Fix__ - a change which fixes an issue
- [ ] __New__ - a change which implements a new feature
- [ ] __Update__ - a change which updates existing functionality

### Platforms
Please select any platforms where your changes have been tested.
- __Desktop__
	- [ ] MacOS
	- [ ] Linux
	- [ ] Windows
- __Mobile__
	- [ ] iOS
	- [ ] Android

### Instructions
Please describe the specific instructions, configurations, and/or test cases necessary to __test and verify__ that your changes work as intended.

## Checklist
Please tick all of the following boxes that are relevant to your changes, and delete those that are not.
- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or modified tests that prove my changes work as intended
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have verified that my latest changes pass CI workflows for testing and linting
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: packages/shared/lib/core/network/actions/getAndUpdateNodeInfo.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { profileManager } from '@lib/wallet'
+import { get } from 'svelte/store'
+import { INodeInfoResponse } from '../interfaces'
+import { updateNodeInfo } from '../stores/node-info.store'
+
+export async function getAndUpdateNodeInfo(): Promise<void> {
+    let nodeInfoResponse: INodeInfoResponse | null
```
Comment (Tuditi) [inline packages/shared/lib/core/network/actions/getAndUpdateNodeInfo.ts:7]: We don't have to explicitely type `null` or `undefined`

File: packages/shared/lib/core/network/actions/getAndUpdateNodeInfo.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    try {
+        nodeInfoResponse = <INodeInfoResponse>(<unknown>await get(profileManager).getNodeInfo())
```
Comment (Tuditi) [inline packages/shared/lib/core/network/actions/getAndUpdateNodeInfo.ts:9]: I guess we'll have to properly adjust the return type of `getNodeInfo()` here. Maybe put a TODO here, so we don't forget?

File: packages/shared/lib/core/network/actions/network-polling.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+let pollInterval
+
+/**
+ * Poll the network status at an interval.
+ */
+export async function pollNetworkStatus(): Promise<void> {
+    await getAndUpdateNodeInfo()
+    /* eslint-disable @typescript-eslint/no-misused-promises */
+    pollInterval = setInterval(async () => getAndUpdateNodeInfo(), DEFAULT_NETWORK_STATUS_POLL_INTERVAL)
```
Comment (Tuditi) [inline packages/shared/lib/core/network/actions/network-polling.ts:12]: This line can be fixed as follows:
`pollInterval = setInterval(() => void getAndUpdateNodeInfo(), DEFAULT_NETWORK_STATUS_POLL_INTERVAL)`

File: packages/shared/lib/core/network/helpers/getNetworkStatusFromNodeInfo.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+export function getNetworkStatusFromNodeInfo(nodeInfo: IStardustNodeInfo | null): INetworkStatus {
+    const timeSinceLastMsInMinutes =
+        (Date.now() - nodeInfo.status.latestMilestone.timestamp * MILLISECONDS_PER_SECOND) /
+        (MILLISECONDS_PER_SECOND * SECONDS_PER_MINUTE)
+
+    let health = NetworkHealth.Down // bad
+    if (timeSinceLastMsInMinutes < 2) {
+        health = NetworkHealth.Operational // good
+    } else if (timeSinceLastMsInMinutes < 5) {
+        health = NetworkHealth.Operational // degraded
```
Comment (Tuditi) [inline packages/shared/lib/core/network/helpers/getNetworkStatusFromNodeInfo.ts:22]: What's the point here? Shouldn't line 22 be `NetworkHealth.Degraded`?

**Review by Tuditi — APPROVED** [review]
Added some comments, but very nice work! :rocket:

File: packages/shared/lib/core/network/actions/getAndUpdateNodeInfo.ts (hunk 1 above)
Comment (nicole-obrien) [reply packages/shared/lib/core/network/actions/getAndUpdateNodeInfo.ts:7]: Done

File: packages/shared/lib/core/network/actions/getAndUpdateNodeInfo.ts (hunk 2 above)
Comment (nicole-obrien) [reply packages/shared/lib/core/network/actions/getAndUpdateNodeInfo.ts:9]: Done

File: packages/shared/lib/core/network/helpers/getNetworkStatusFromNodeInfo.ts (hunk 4 above)
Comment (nicole-obrien) [reply packages/shared/lib/core/network/helpers/getNetworkStatusFromNodeInfo.ts:22]: done

File: packages/shared/lib/core/network/actions/network-polling.ts (hunk 3 above)
Comment (nicole-obrien) [reply packages/shared/lib/core/network/actions/network-polling.ts:12]: done

**Pull request merged**